s390x/ppc64le have CMake Error at nativeaot/Runtime/Full/CMakeLists.txt

The cross-builds we have for s390x and ppc64le are failing:
```
    CMake Error at nativeaot/Runtime/Full/CMakeLists.txt:187 (add_library):
      Cannot find source file:
    
        /home/tester/workdir/dotnet/src/runtime/src/coreclr/runtime/s390x/AllocFast.S
    
    
    CMake Error at nativeaot/Runtime/Full/CMakeLists.txt:187 (add_library):
      No SOURCES given to target: Runtime.AsmHelpers
```

Probably this regressed in https://github.com/dotnet/dotnet/pull/7004.

https://github.com/dotnet/dotnet/blob/d8addc1562ad98c7cab19ed654e9caabd1112d87/src/arcade/eng/common/native/NativeAotSupported.props#L13-L17

vs

https://github.com/dotnet/dotnet/blob/e9cde69d4ee92243ccd47d9dd0b2ffd5f9cf4c3c/src/runtime/eng/Subsets.props#L51

cc @am11 @akoeplinger @ViktorHofer @omajid

--- a/src/arcade/eng/common/native/NativeAotSupported.props
+++ b/src/arcade/eng/common/native/NativeAotSupported.props
@@ -13,6 +13,8 @@
     <!-- Reject unsupported architectures via RID suffix match -->
     <_NativeAotSupportedArch Condition="
       '$(TargetArchitecture)' != 'wasm' and
+      '$(TargetArchitecture)' != 's390x' and
+      '$(TargetArchitecture)' != 'ppc64le' and
       ('$(TargetArchitecture)' != 'x86' or '$(TargetOS)' == 'windows')
     ">true</_NativeAotSupportedArch>
 

--- a/src/aspnetcore/eng/common/native/NativeAotSupported.props
+++ b/src/aspnetcore/eng/common/native/NativeAotSupported.props
@@ -13,6 +13,8 @@
     <!-- Reject unsupported architectures via RID suffix match -->
     <_NativeAotSupportedArch Condition="
       '$(TargetArchitecture)' != 'wasm' and
+      '$(TargetArchitecture)' != 's390x' and
+      '$(TargetArchitecture)' != 'ppc64le' and
       ('$(TargetArchitecture)' != 'x86' or '$(TargetOS)' == 'windows')
     ">true</_NativeAotSupportedArch>
 

--- a/src/efcore/eng/common/native/NativeAotSupported.props
+++ b/src/efcore/eng/common/native/NativeAotSupported.props
@@ -13,6 +13,8 @@
     <!-- Reject unsupported architectures via RID suffix match -->
     <_NativeAotSupportedArch Condition="
       '$(TargetArchitecture)' != 'wasm' and
+      '$(TargetArchitecture)' != 's390x' and
+      '$(TargetArchitecture)' != 'ppc64le' and
       ('$(TargetArchitecture)' != 'x86' or '$(TargetOS)' == 'windows')
     ">true</_NativeAotSupportedArch>
 

--- a/src/runtime/eng/common/native/NativeAotSupported.props
+++ b/src/runtime/eng/common/native/NativeAotSupported.props
@@ -13,6 +13,8 @@
     <!-- Reject unsupported architectures via RID suffix match -->
     <_NativeAotSupportedArch Condition="
       '$(TargetArchitecture)' != 'wasm' and
+      '$(TargetArchitecture)' != 's390x' and
+      '$(TargetArchitecture)' != 'ppc64le' and
       ('$(TargetArchitecture)' != 'x86' or '$(TargetOS)' == 'windows')
     ">true</_NativeAotSupportedArch>
 

--- a/src/runtime/src/coreclr/CMakeLists.txt
+++ b/src/runtime/src/coreclr/CMakeLists.txt
@@ -156,7 +156,7 @@ endif ()
 if(NOT CLR_CROSS_COMPONENTS_BUILD)
   # NativeAOT is buildable for all CoreCLR-supported configurations except a known reject list.
   # Keep this in sync with the NativeAotSupported reject list in eng/common/native/NativeAotSupported.props.
-  if(NOT (CLR_CMAKE_HOST_ARCH_WASM OR (CLR_CMAKE_HOST_ARCH_I386 AND NOT CLR_CMAKE_HOST_WIN32)))
+  if(NOT (CLR_CMAKE_HOST_ARCH_WASM OR CLR_CMAKE_HOST_ARCH_S390X OR CLR_CMAKE_HOST_ARCH_POWERPC64 OR (CLR_CMAKE_HOST_ARCH_I386 AND NOT CLR_CMAKE_HOST_WIN32)))
     add_subdirectory(nativeaot)
   endif()
 endif(NOT CLR_CROSS_COMPONENTS_BUILD)

--- a/src/sdk/eng/common/native/NativeAotSupported.props
+++ b/src/sdk/eng/common/native/NativeAotSupported.props
@@ -13,6 +13,8 @@
     <!-- Reject unsupported architectures via RID suffix match -->
     <_NativeAotSupportedArch Condition="
       '$(TargetArchitecture)' != 'wasm' and
+      '$(TargetArchitecture)' != 's390x' and
+      '$(TargetArchitecture)' != 'ppc64le' and
       ('$(TargetArchitecture)' != 'x86' or '$(TargetOS)' == 'windows')
     ">true</_NativeAotSupportedArch>
 

